addons/docs/src/frameworks/common/inferControls.ts

```diff
@@ -4,21 +4,27 @@ import { Control } from '@storybook/components';
 import { SBEnumType } from '../../lib/sbtypes';
 
 const inferControl = (argType: ArgType): Control => {
-  if (!argType.type) {
+  const { type } = argType;
+  if (!type) {
     // console.log('no sbtype', { argType });
     return null;
   }
-  switch (argType.type.name) {
-    case 'array':
+  switch (type.name) {
+    case 'array': {
+      const { value } = type;
+      if (value?.name && ['object', 'other'].includes(value.name)) {
+        return { type: 'object' };
```

Comment:
I think we discussed doing some type checks when you edit the control to make sure that the object you entered is not only valid js but also an array. What are you thinking about that @shilman ?

Reply:
Good point. Updated with "must be array" validation, but not sure whether using a functional validator will bite us later when we try to put this in the manager